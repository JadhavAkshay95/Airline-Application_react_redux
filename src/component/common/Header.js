import React from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Logo from './Logo';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom/cjs/react-router-dom';
import Button from 'react-bootstrap/Button';
import * as LoginAction from './../../redux/actions/loginAction';
import PropTypes from 'prop-types';

export const Header = ({ currentUser, logout }) => {
  const logoutUser = () => {
    logout({});
  };

  return (
    <Container className="header" fluid="true">
      <div className="row-header">
        <Col>
          <Logo />
        </Col>
        <Col style={{ textAlign: 'end' }}>
          {currentUser && (
            <>
              <div
                style={{
                  display: 'inline',
                  margin: ' 10px',
                  color: 'white'
                }}
              >
                <span>
                  Hello {currentUser.role} {currentUser.username}
                </span>
              </div>

              <Link to="/">
                <Button variant="secondary" onClick={logoutUser}>
                  Logout
                </Button>
              </Link>
            </>
          )}
        </Col>
      </div>
    </Container>
  );
};

const mapStateToProps = state => {
  return {
    currentUser: Array.isArray(state.loginReducer) ? null : state.loginReducer
  };
};
const mapDispatchToProps = {
  logout: LoginAction.logout
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
Header.propTypes = { currentUser: PropTypes.object, logout: PropTypes.func };
