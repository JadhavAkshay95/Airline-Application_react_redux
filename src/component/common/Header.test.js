import React from 'react'
import { shallow } from 'enzyme'
import { Header } from './Header'

const renderLoginModal = args => {
  const defaultProps = { history: {}, currentUser: {}, logout: () => {} }
  const props = { ...defaultProps, ...args }
  return shallow(<Header {...props} />)
}

describe('Render Header component', () => {
  it('Render Header ', () => {
    const wrapper = renderLoginModal()
    expect(wrapper).toMatchSnapshot()
  })
})
