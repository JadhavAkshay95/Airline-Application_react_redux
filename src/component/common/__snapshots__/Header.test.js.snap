// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header component Render Header  1`] = `ShallowWrapper {}`;
